tests/page-objects/SurveyPage.ts

```diff
@@ -2,52 +2,121 @@ import { Selector, t } from 'testcafe';
+        await t.expect(question.exists).ok();
+        const input = question.prevSibling('.select-field__control').find('input');
+        await t.typeText(input, answer).pressKey('enter');
     }
 
-    private getAnswerInput(question: 1 | 2 | 3): Selector {
-        switch (question) {
-            case 1:
-                return this.answer1Input;
-            case 2:
-                return this.answer2Input;
-            case 3:
-                return this.answer3Input;
-        }
+    private async getAnswerFromSelectField(question: Selector): Promise<string> {
+        await t.expect(question.exists).ok();
+        return question.value;
+    }
+
+    private async populatePage1(): Promise<void> {
+        await this.assertOnPage1();
+        await this.answerRadioField(this.submittingAs, 'first-author');
+        await this.answerRadioField(this.independentResearcher, 'yes');
+        await t.expect(this.independentResearcherYear.exists).ok();  
+        await this.answerTextField(this.independentResearcherYear, '1990');
+        await t.expect(await this.getAnswerFromTextField(this.independentResearcherYear)).eql('1990');
+    }
+
+    private async populatePage2(): Promise<void> {
+        await this.assertOnPage2();
+        await this.answerRadioField(this.genderIdentity, 'self-describe');
+        await t.expect(this.genderSelfDescribe.exists).ok();  
+        await this.answerTextField(this.genderSelfDescribe, 'popsicle');
+        await this.answerSelectField(this.countryOfResidence, 'United Kingdom');
+        await t.expect(await this.getAnswerFromSelectField(this.countryOfResidence)).eql('GB');
+        await clickSelector('#secondCountryOfResidenceToggle * .content-toggle__toggle-btn');
+        await t.expect(this.secondCountryOfResidence.exists).ok(); 
+        await this.answerSelectField(this.secondCountryOfResidence, 'Albania');
+        await t.expect(await this.getAnswerFromSelectField(this.secondCountryOfResidence)).eql('AL');
+        await this.answerRadioField(this.countryIndentifyAs, 'yes');
+        await this.raceOrEthnicity(this.independentResearcherYear, 'jedi');
     }
 
     public async populateForm(): Promise<void> {
-        await this.answerQuestion(1);
-        await this.answerQuestion(2);
-        await this.answerQuestion(3);
+        await this.assertOnPage();
+        await this.populatePage1();
+        await t.expect(Selector(this.doneSkipButton).exists).ok();
+        await t.expect(Selector(this.doneSkipButton).visible).ok();
+        await clickSelector(this.doneSkipButton);
+        await t.wait(500);
+        await this.populatePage2();
     }
 
-    public async skipOrFinish(): Promise<void> {
+    public async completeSurvey(): Promise<void> {
         await t.expect(Selector(this.doneSkipButton).exists).ok();
         await t.expect(Selector(this.doneSkipButton).visible).ok();
         await clickSelector(this.doneSkipButton);
```

Comment:
These 3 lines could be replaced with a call to the private function suggested above

Reply:
*4 lines, looks like the wait is below this comment